### ISSUE TITLE
Error when checkov is trying to list nested modules

### ISSUE THREAD

**Issue by dmnord** [bridgecrewio/checkov#5443]
**Description**
The error happens when nested module is checked during the run. If this module is commented in the code then everything is going fine, and vise versa if this module is uncommented out in the code then the issue happens. This module is being called and used in different places across our Terraform code and only one single place in the code -- when a nested call is used -- produces this error. Though, this module is always nested in different places within a code.

I tested this flag (CHECKOV_ENABLE_MODULES_FOREACH_HANDLING) setting it to false and suddenly I realized that the issue was solving when this parameter was used. Unfortunately, I don't see an issue but some items from the resulting list has disappeared.

**Example**
The environment has huge number of modules and nested ones across the code with for_each and count .
Example might look like the following:
```terraform
# a call of the parent module
module "gke_cluster" {
  for_each = var.orchestration_gke_clusters
  source   = "../../modules/gcp/gke_cluster"
  # set of parameters
  # ...
  gke_firewall_rules  = var.gke_cluster_firewall_rules_list
  # ...
}
```
```terraform
# nested module defined within a parent module described above
module "firewall_rules" {
  count                           = length(var.gke_firewall_rules)
  source                          = "../firewall_rule"
  # set of parameters
  # ...
}
```

**Exception Trace**
```sh
# checkov --framework=terraform -d .
2023-08-15 08:11:43,038 [MainThread  ] [ERROR]  Exception traceback:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/checkov/main.py", line 470, in run
    self.scan_reports = runner_registry.run(
  File "/usr/local/lib/python3.10/site-packages/checkov/common/runners/runner_registry.py", line 119, in run
    self.runners[0].run(root_folder, external_checks_dir=external_checks_dir, files=files,
  File "/usr/local/lib/python3.10/site-packages/checkov/terraform/runner.py", line 171, in run
    self.check_tf_definition(report, root_folder, runner_filter, collect_skip_comments)
  File "/usr/local/lib/python3.10/site-packages/checkov/terraform/runner.py", line 351, in check_tf_definition
    self.push_skipped_checks_down_from_modules(self.context)
  File "/usr/local/lib/python3.10/site-packages/checkov/terraform/runner.py", line 595, in push_skipped_checks_down_from_modules
    self.push_skipped_checks_down(definition_context, skipped_checks, resolved_paths)
  File "/usr/local/lib/python3.10/site-packages/checkov/terraform/runner.py", line 618, in push_skipped_checks_down
    module_context = next(m for m in self.definitions.get(resolved_paths[ind]).get(block_type) if module_name in m)
AttributeError: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/usr/local/bin/checkov", line 9, in <module>
    sys.exit(Checkov().run())
  File "/usr/local/lib/python3.10/site-packages/checkov/main.py", line 470, in run
    self.scan_reports = runner_registry.run(
  File "/usr/local/lib/python3.10/site-packages/checkov/common/runners/runner_registry.py", line 119, in run
    self.runners[0].run(root_folder, external_checks_dir=external_checks_dir, files=files,
  File "/usr/local/lib/python3.10/site-packages/checkov/terraform/runner.py", line 171, in run
    self.check_tf_definition(report, root_folder, runner_filter, collect_skip_comments)
  File "/usr/local/lib/python3.10/site-packages/checkov/terraform/runner.py", line 351, in check_tf_definition
    self.push_skipped_checks_down_from_modules(self.context)
  File "/usr/local/lib/python3.10/site-packages/checkov/terraform/runner.py", line 595, in push_skipped_checks_down_from_modules
    self.push_skipped_checks_down(definition_context, skipped_checks, resolved_paths)
  File "/usr/local/lib/python3.10/site-packages/checkov/terraform/runner.py", line 618, in push_skipped_checks_down
    module_context = next(m for m in self.definitions.get(resolved_paths[ind]).get(block_type) if module_name in m)
AttributeError: 'NoneType' object has no attribute 'get'
```

**Desktop (please complete the following information):**
 - OS: Linux
 - Python Version: 3.10.x, 3.11.x
 - Checkov Version: 2.3.365

**Comment by jbrule** [comment]
We use a lot of nested modules and count, for_each too. I have been relying on plan inspection with repo enrichment as I just get too many erroneous findings from tf scans. Have you tried plan scanning?

**Comment by dmnord** [comment]
So, the issue has disappeared since this [PR #5463](https://github.com/bridgecrewio/checkov/pull/5463) is implemented and version 2.4.5 is released.
